modules/xfem/src/materials/GeometricCutSwitchingMaterial.C

```diff
@@ -0,0 +1,103 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GeometricCutSwitchingMaterial.h"
+
+registerMooseObject("XFEMApp", GeometricCutSwitchingMaterialReal);
+registerMooseObject("XFEMApp", GeometricCutSwitchingMaterialRankTwoTensor);
+registerMooseObject("XFEMApp", GeometricCutSwitchingMaterialRankThreeTensor);
+registerMooseObject("XFEMApp", GeometricCutSwitchingMaterialRankFourTensor);
+
+registerMooseObject("XFEMApp", ADGeometricCutSwitchingMaterialReal);
+registerMooseObject("XFEMApp", ADGeometricCutSwitchingMaterialRankTwoTensor);
+registerMooseObject("XFEMApp", ADGeometricCutSwitchingMaterialRankThreeTensor);
+registerMooseObject("XFEMApp", ADGeometricCutSwitchingMaterialRankFourTensor);
+
+template <typename T, bool is_ad>
+InputParameters
+GeometricCutSwitchingMaterialTempl<T, is_ad>::validParams()
+{
+  InputParameters params = Material::validParams();
+  params.addClassDescription("Switch the material property based on the GeometricCutSubdomainID.");
+  params.addRequiredParam<UserObjectName>("geometric_cut_userobject",
+                                          "The geometric cut userobject");
+  params.addRequiredParam<std::vector<GeometricCutSubdomainID>>(
+      "base_name_keys",
```

Comment:
I think it might be clearer to call these parameters `geometric_cut_subdomain_ids` and `base_names`. We can discuss, though, if you don't like those suggestions.

Reply:
`geometric_cut_subdomain_ids` and `base_names` are better. Again, these parameter names were carried over from the previous design where the key/value naming made more sense.